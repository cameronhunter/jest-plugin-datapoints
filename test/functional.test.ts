import * as globals from '@jest/globals';

describe('Using @jest/globals package', () => {
  describe('test', () => {
    const suite = globals.test.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  describe('it', () => {
    const suite = globals.it.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  /**
   * These tests should be marked as skipped in the test suite.
   */
  describe('xtest', () => {
    const suite = globals.xtest.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  /**
   * These tests should be marked as skipped in the test suite.
   */
  describe('xit', () => {
    const suite = globals.xit.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });
});

describe('Using global variable', () => {
  describe('test', () => {
    const suite = test.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  describe('it', () => {
    const suite = it.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  /**
   * These tests should be marked as skipped in the test suite.
   */
  describe('xtest', () => {
    const suite = xtest.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });

  /**
   * These tests should be marked as skipped in the test suite.
   */
  describe('xit', () => {
    const suite = xit.datapoints({
      one: [1, 2, 3],
      two: ['a', 'b', 'c']
    });

    suite('Testcase: $one, $two', ({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });

    suite(({ one, two }) => {
      expect(one).toEqual(expect.any(Number));
      expect(two).toEqual(expect.any(String));
    });
  });
});
