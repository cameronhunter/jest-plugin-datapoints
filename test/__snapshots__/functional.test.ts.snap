// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot {"a":false,"b":false} 1`] = `false`;

exports[`snapshot {"a":false,"b":true} 1`] = `false`;

exports[`snapshot {"a":true,"b":false} 1`] = `false`;

exports[`snapshot {"a":true,"b":true} 1`] = `true`;
